parser/src/__tests__/parseResult.test.ts

```diff
@@ -0,0 +1,104 @@
+import { groupByUniqueName } from "..";
+import { parseMachinesFromFile } from "../parseMachinesFromFile";
+
+describe("MachineParseResult", () => {
+  it("Should let you get a state node by path", () => {
+    const result = parseMachinesFromFile(`
+      createMachine({
+        states: {
+          a: {},
+          b: {
+            states: {
+              b1: {}
+            }
+          }
+        },
+      })
+    `);
+
+    const machine = result.machines[0];
+
+    expect(machine.getAllStateNodes()).toHaveLength(4);
+
+    const aNode = machine.getStateNodeByPath(["a"]);
+
+    expect(aNode?.path).toEqual(["a"]);
+
+    const b1Node = machine.getStateNodeByPath(["b", "b1"]);
```

Comment:
q: what this is useful for?

Reply:
Interesting. Could you describe this in more detail? Like what's the exact user story for this?

Reply:
The way we store the AST uses a recursive structure, similar to XState's config itself:

```
// pseudo-code
definition: {
  states: {
    a: {
      states: {}
    }
  }
}
```

This is a useful helper for when you need to traverse deep in the state. It also means we can directly interop between XState's paths and the AST.

The user story is:

WHEN a user CMD-clicks a target
THEN they should be able to jump to the targeted node via jump-to-definition